sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioUtils.scala

```diff
@@ -16,92 +16,267 @@
 
 package com.nvidia.spark.rapids
 
+import java.io.FileNotFoundException
+
+import scala.io.Source
+import scala.sys.process.{Process, ProcessLogger}
+
 import org.apache.hadoop.fs.Path
 
+import org.apache.spark.internal.Logging
 import org.apache.spark.sql.catalyst.expressions.{Expression, PlanExpression}
 import org.apache.spark.sql.execution.datasources.{FileIndex, HadoopFsRelation, InMemoryFileIndex}
 import org.apache.spark.sql.execution.datasources.rapids.GpuPartitioningUtils
 
-object AlluxioUtils {
+object AlluxioUtils extends Logging {
+  val mountedBuckets: scala.collection.mutable.Map[String, String] = scala.collection.mutable.Map()
+  var alluxioCmd: Seq[String] = null
+  var alluxioMasterHost: Option[String] = None
+  var alluxio_home: String = "/opt/alluxio-2.8.0"
+  var isInit: Boolean = false
+
+  // Read out alluxio.master.hostname, alluxio.master.rpc.port
+  // from Alluxio's conf alluxio-site.properties
+  // We require an environment variable "ALLUXIO_HOME"
+  // This function will only read once from ALLUXIO/conf.
+  private def initAlluxioInfo(conf: RapidsConf): Unit = {
+    this.synchronized {
+      alluxio_home = scala.util.Properties.envOrElse("ALLUXIO_HOME", "/opt/alluxio-2.8.0")
```

Comment:
is there any reason why not to put the alluxio_home/alluxio_Cmd into the `if (!isInit)` block?

Reply:
Just consider for the case that user changed the spark.rapids.alluxio.cmd at runtime.
So, it can be read in and use the new command. Mainly easier for debug case.